homeassistant/components/esphome/ffmpeg_proxy.py

```diff
@@ -230,18 +234,14 @@ async def _write_ffmpeg_data(
             raise  # don't log error
         except:
             _LOGGER.exception("Unexpected error during ffmpeg conversion")
-
-            # Process did not exit successfully
-            stderr_text = ""
-            while line := await proc.stderr.readline():
-                stderr_text += line.decode()
-            _LOGGER.error("FFmpeg output: %s", stderr_text)
-
             raise
         finally:
             # Allow conversion info to be removed
             self.convert_info.is_finished = True
 
+            # stop dumping ffmpeg stderr task
+            stderr_task.cancel()
```

Comment:
If the task throws an exception it will never be retrieved because its not awaited after cancellation

Reply:
nevermind, calling `.cancel()` will suppress the traceback for an un-retrieved exception
https://github.com/python/cpython/blob/71ede1142ddad2d31cc966b8fe4a5aff664f4d53/Lib/asyncio/futures.py#L149
https://github.com/python/cpython/blob/71ede1142ddad2d31cc966b8fe4a5aff664f4d53/Lib/asyncio/futures.py#L91

Reply:
we've discussed this on discord - or better said @bdraco checked the cPython code and it figured out, that `.cancel()` meanwhile suppress any saved exception of the task